### PR TITLE
Persisting

### PR DESCRIPTION
**Pull request by fjtirado**
Adding setDataDir configuration through quarkus.embedded.postgresql.data.dir property
Env variable QUARKUS_EMBEDDED_POSTGRESQL_DATA_DIR
It also add support for datasource.

### CODE REVIEW

File: runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLRecorder.java [hunk 1]
```diff
@@ -1,40 +1,33 @@
...
 public class EmbeddedPostgreSQLRecorder {
 
-    private static final Logger log = Logger.getLogger(EmbeddedPostgreSQLRecorder.class);
+    private static final Logger logger = Logger.getLogger(EmbeddedPostgreSQLRecorder.class);
 
     public RuntimeValue<Integer> startPostgres() throws IOException {
-        EmbeddedPostgres pg = EmbeddedPostgres.start();
-        log.infov("Embedded Postgres started at port \"{0,number,#}\" with database \"{1}\", user \"{2}\" and password \"{3}\"",
+        Builder builder = EmbeddedPostgres.builder();
+        ConfigProvider.getConfig().getOptionalValue("quarkus.embedded.postgresql.data.dir", String.class).ifPresent(path -> {
```
Comment (fjtirado) [inline runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLRecorder.java:23]: I tried injecting EmbeddedPostgreSQLConfiguration as field of the recorder class  but it does not work (it is null), so accesing the property programatically. 
I manually tested it takes the env value at runtime.

File: runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package io.quarkiverse.embedded.postgresql;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.RUN_TIME, prefix = "quarkus.embedded.postgresql")
```
Comment (cristianonicolai) [inline runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLConfiguration.java:9]: Use _name=embedded.postgresql_ instead. Prefix is default to _quarkus_

File: runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLRecorder.java (hunk 1 above)
Comment (cristianonicolai) [reply runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLRecorder.java:23]: Im not sure if possible but could you just inject the config object instead?

Comment (fjtirado) [reply runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLRecorder.java:23]: Yes, that can be probably done, Ill try

Comment (fjtirado) [reply runtime/src/main/java/io/quarkiverse/embedded/postgresql/EmbeddedPostgreSQLRecorder.java:23]: No, its not working (;, Ill kept it that way

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**